File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Order.java
package com.example.grts;

import viaduct.engine.api.EngineObjectData;
import viaduct.engine.api.NodeReference;
import viaduct.engine.api.RootFieldReference;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.internal.NodeObjectBase;
import viaduct.java.api.internal.ObjectBase;
import viaduct.java.api.internal.OutputBuilderTypeChecker;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.RootObjectField;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.Arguments;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

@SuppressWarnings("MissingOverride")
public class Order extends NodeObjectBase implements Node, Auditable, Timestamped, SearchHit {

    public static final Type<Order> Reflection = Type.ofClass(Order.class);

    public static final class Fields implements TypeFields<Order> {
        private Fields() {}

        public static final Field<Order> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Order> id =
                                Field.of("id", Reflection);

                public static final CompositeField<Order, OrderStatus> status =
                                CompositeField.of("status", Reflection, OrderStatus.Reflection);

                public static final CompositeField<Order, Money> total =
                                CompositeField.of("total", Reflection, Money.Reflection);

                public static final Field<Order> createdAt =
                                Field.of("createdAt", Reflection);

                public static final Field<Order> updatedAt =
                                Field.of("updatedAt", Reflection);

                public static final Field<Order> auditTrail =
                                Field.of("auditTrail", Reflection);

                public static final CompositeField<Order, User> buyer =
                                CompositeField.of("buyer", Reflection, User.Reflection);

    }

    public Order(InternalContext context, EngineObjectData.Sync data) {
        super(context, data);
    }

    @SuppressWarnings("UnusedMethod")
    private Order(InternalContext context, Map<String, Object> data) {
        super(context, data, "Order");
    }

    private Order(InternalContext context, ObjectBase base, Map<String, Object> data) {
        super(context, base, data, "Order");
    }

    public Order(InternalContext context, RootFieldReference rootFieldReference) {
        super(context, rootFieldReference);
    }

    public Order(InternalContext context, NodeReference nodeReference) {
        super(context, nodeReference);
    }

        public GlobalID<Order> getIdOrThrow(String alias) {
            return fetchGlobalID("id", alias);
        }

        public GlobalID<Order> getIdOrThrow() {
            return fetchGlobalID("id", null);
        }

        public GlobalID<Order> getId(String alias) {
            return nullOnDataFailure(() -> fetchGlobalID("id", alias));
        }

        public GlobalID<Order> getId() {
            return nullOnDataFailure(() -> fetchGlobalID("id", null));
        }

        public OrderStatus getStatusOrThrow(String alias) {
            return fetchEnum("status", alias, OrderStatus.class);
        }

        public OrderStatus getStatusOrThrow() {
            return fetchEnum("status", null, OrderStatus.class);
        }

        public OrderStatus getStatus(String alias) {
            return nullOnDataFailure(() -> fetchEnum("status", alias, OrderStatus.class));
        }

        public OrderStatus getStatus() {
            return nullOnDataFailure(() -> fetchEnum("status", null, OrderStatus.class));
        }

        public Money getTotalOrThrow(String alias) {
            return fetchObject("total", alias, Money.class, Money::new);
        }

        public Money getTotalOrThrow() {
            return fetchObject("total", null, Money.class, Money::new);
        }

        public Money getTotal(String alias) {
            return nullOnDataFailure(() -> fetchObject("total", alias, Money.class, Money::new));
        }

        public Money getTotal() {
            return nullOnDataFailure(() -> fetchObject("total", null, Money.class, Money::new));
        }

        public String getCreatedAtOrThrow(String alias) {
            return fetchScalar("createdAt", alias);
        }

        public String getCreatedAtOrThrow() {
            return fetchScalar("createdAt", null);
        }

        public String getCreatedAt(String alias) {
            return nullOnDataFailure(() -> fetchScalar("createdAt", alias));
        }

        public String getCreatedAt() {
            return nullOnDataFailure(() -> fetchScalar("createdAt", null));
        }

        public String getUpdatedAtOrThrow(String alias) {
            return fetchScalar("updatedAt", alias);
        }

        public String getUpdatedAtOrThrow() {
            return fetchScalar("updatedAt", null);
        }

        public String getUpdatedAt(String alias) {
            return nullOnDataFailure(() -> fetchScalar("updatedAt", alias));
        }

        public String getUpdatedAt() {
            return nullOnDataFailure(() -> fetchScalar("updatedAt", null));
        }

        public List<String> getAuditTrailOrThrow(String alias) {
            return fetchScalarList("auditTrail", alias);
        }

        public List<String> getAuditTrailOrThrow() {
            return fetchScalarList("auditTrail", null);
        }

        public List<String> getAuditTrail(String alias) {
            return nullOnDataFailure(() -> fetchScalarList("auditTrail", alias));
        }

        public List<String> getAuditTrail() {
            return nullOnDataFailure(() -> fetchScalarList("auditTrail", null));
        }

        public User getBuyerOrThrow(String alias) {
            return fetchObject("buyer", alias, User.class, User::new);
        }

        public User getBuyerOrThrow() {
            return fetchObject("buyer", null, User.class, User::new);
        }

        public User getBuyer(String alias) {
            return nullOnDataFailure(() -> fetchObject("buyer", alias, User.class, User::new));
        }

        public User getBuyer() {
            return nullOnDataFailure(() -> fetchObject("buyer", null, User.class, User::new));
        }


    public Builder toBuilder() {
        return new Builder(__context(), toBuilderBase());
    }

    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context), null);
    }

    public static class Builder {
        private final InternalContext __context;
        private final ObjectBase __base;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context, ObjectBase __base) {
            this.__context = __context;
            this.__base = __base;
        }

                public Builder id(GlobalID<Order> id) {
                    id = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "id",
                            null,
                            id);
                    data.put("id", id == null ? null : __context.getGlobalIDCodec().serialize(id.getType().getName(), id.getInternalID()));
        return this;
                }

                public Builder status(OrderStatus status) {
                    status = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "status",
                            OrderStatus.class,
                            status);
                    data.put("status", status);
        return this;
                }

                public Builder total(Money total) {
                    total = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "total",
                            Money.class,
                            total);
                    data.put("total", total);
        return this;
                }

                public Builder createdAt(String createdAt) {
                    createdAt = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "createdAt",
                            null,
                            createdAt);
                    data.put("createdAt", createdAt);
        return this;
                }

                public Builder updatedAt(String updatedAt) {
                    updatedAt = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "updatedAt",
                            null,
                            updatedAt);
                    data.put("updatedAt", updatedAt);
        return this;
                }

                public Builder auditTrail(List<String> auditTrail) {
                    auditTrail = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "auditTrail",
                            null,
                            auditTrail);
                    data.put("auditTrail", auditTrail);
        return this;
                }

                public Builder buyer(User buyer) {
                    buyer = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Order",
                            "buyer",
                            User.class,
                            buyer);
                    data.put("buyer", buyer);
        return this;
                }


        public Order build() {
            return new Order(__context, __base, new LinkedHashMap<>(data));
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Order.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InternalContext
import viaduct.api.internal.ObjectBase
import viaduct.engine.api.EngineObject
import viaduct.engine.api.EngineObjectData

@OptIn(InternalApi::class)
class Order(context: InternalContext, engineObject: EngineObject)
    : ObjectBase(context, engineObject), viaduct.api.types.Object,viaduct.api.grts.Node,viaduct.api.grts.Auditable,viaduct.api.grts.Timestamped,viaduct.api.grts.SearchHit,viaduct.api.types.NodeObject
{
          final override fun getIdOrThrow(alias: String?): viaduct.api.globalid.GlobalID<viaduct.api.grts.Order> = TODO()
          final override fun getIdOrThrow(): viaduct.api.globalid.GlobalID<viaduct.api.grts.Order> = TODO()
          final override fun getId(alias: String?): viaduct.api.globalid.GlobalID<viaduct.api.grts.Order>? = TODO()
          final override fun getId(): viaduct.api.globalid.GlobalID<viaduct.api.grts.Order>? = TODO()

     fun getStatusOrThrow(alias: String?): viaduct.api.grts.OrderStatus = TODO()
     fun getStatusOrThrow(): viaduct.api.grts.OrderStatus = TODO()
     fun getStatus(alias: String?): viaduct.api.grts.OrderStatus? = TODO()
     fun getStatus(): viaduct.api.grts.OrderStatus? = TODO()

     fun getTotalOrThrow(alias: String?): viaduct.api.grts.Money = TODO()
     fun getTotalOrThrow(): viaduct.api.grts.Money = TODO()
     fun getTotal(alias: String?): viaduct.api.grts.Money? = TODO()
     fun getTotal(): viaduct.api.grts.Money? = TODO()

          final override fun getCreatedAtOrThrow(alias: String?): kotlin.String = TODO()
          final override fun getCreatedAtOrThrow(): kotlin.String = TODO()
          final override fun getCreatedAt(alias: String?): kotlin.String? = TODO()
          final override fun getCreatedAt(): kotlin.String? = TODO()

          final override fun getUpdatedAtOrThrow(alias: String?): kotlin.String? = TODO()
          final override fun getUpdatedAtOrThrow(): kotlin.String? = TODO()
          final override fun getUpdatedAt(alias: String?): kotlin.String? = TODO()
          final override fun getUpdatedAt(): kotlin.String? = TODO()

          final override fun getAuditTrailOrThrow(alias: String?): kotlin.collections.List<kotlin.String>? = TODO()
          final override fun getAuditTrailOrThrow(): kotlin.collections.List<kotlin.String>? = TODO()
          final override fun getAuditTrail(alias: String?): kotlin.collections.List<kotlin.String>? = TODO()
          final override fun getAuditTrail(): kotlin.collections.List<kotlin.String>? = TODO()

     fun getBuyerOrThrow(alias: String?): viaduct.api.grts.User = TODO()
     fun getBuyerOrThrow(): viaduct.api.grts.User = TODO()
     fun getBuyer(alias: String?): viaduct.api.grts.User? = TODO()
     fun getBuyer(): viaduct.api.grts.User? = TODO()


    fun toBuilder(): Builder =
        Builder(__context, __engineObject.type, toBuilderEOD())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Order =
            Builder(context).apply(block).build()
    }

    class Builder : ObjectBase.Builder<Order> {
        constructor(context: ExecutionContext)
            : super(
                context as InternalContext,
                TODO() as graphql.schema.GraphQLObjectType,
                null
            )

        internal constructor(
            context: InternalContext,
            type: graphql.schema.GraphQLObjectType,
            baseEngineObjectData: EngineObjectData.Sync
        ) : super(context, type, baseEngineObjectData)

                  fun id(value: viaduct.api.globalid.GlobalID<viaduct.api.grts.Order>): Builder = TODO()

                  fun status(value: viaduct.api.grts.OrderStatus): Builder = TODO()

                  fun total(value: viaduct.api.grts.Money): Builder = TODO()

                  fun createdAt(value: kotlin.String): Builder = TODO()

                  fun updatedAt(value: kotlin.String?): Builder = TODO()

                  fun auditTrail(value: kotlin.collections.List<kotlin.String>?): Builder = TODO()

                  fun buyer(value: viaduct.api.grts.User): Builder = TODO()


        final override fun build(): Order = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Order> {
        override final val name = "Order"
        override final val kcls = viaduct.api.grts.Order::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Order> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Order> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Order.Reflection)

            final val id: viaduct.api.reflect.Field<viaduct.api.grts.Order> =
                viaduct.api.internal.FieldImpl("id", viaduct.api.grts.Order.Reflection)

            final val status: viaduct.api.reflect.CompositeField<viaduct.api.grts.Order, viaduct.api.grts.OrderStatus> =
                viaduct.api.internal.CompositeFieldImpl("status", viaduct.api.grts.Order.Reflection, viaduct.api.grts.OrderStatus.Reflection)

            final val total: viaduct.api.reflect.CompositeField<viaduct.api.grts.Order, viaduct.api.grts.Money> =
                viaduct.api.internal.CompositeFieldImpl("total", viaduct.api.grts.Order.Reflection, viaduct.api.grts.Money.Reflection)

            final val createdAt: viaduct.api.reflect.Field<viaduct.api.grts.Order> =
                viaduct.api.internal.FieldImpl("createdAt", viaduct.api.grts.Order.Reflection)

            final val updatedAt: viaduct.api.reflect.Field<viaduct.api.grts.Order> =
                viaduct.api.internal.FieldImpl("updatedAt", viaduct.api.grts.Order.Reflection)

            final val auditTrail: viaduct.api.reflect.Field<viaduct.api.grts.Order> =
                viaduct.api.internal.FieldImpl("auditTrail", viaduct.api.grts.Order.Reflection)

            final val buyer: viaduct.api.reflect.CompositeField<viaduct.api.grts.Order, viaduct.api.grts.User> =
                viaduct.api.internal.CompositeFieldImpl("buyer", viaduct.api.grts.Order.Reflection, viaduct.api.grts.User.Reflection)

    }

}